src/Parser/Hunter/Marksmanship/Modules/Talents/TrueAim.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox from 'Main/StatisticBox';
+
+class TrueAim extends Module {
+  static dependencies = {
+    combatants: Combatants,
+    enemies: Enemies,
+  };
+
+  _currentStacks = 0;
+  startOfMaxStacks = 0;
+  timeAtMaxStacks = 0;
+
+  //starts -1 because the stacks drop at end of combat, but that shouldn't count as a time where it dropped
+  timesDropped = -1;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.TRUE_AIM_TALENT.id);
+  }
+
+  on_byPlayer_applydebuffstack(event) {
+    const debuffId = event.ability.guid;
+    if (debuffId !== SPELLS.TRUE_AIM_DEBUFF.id) {
+      return;
+    }
+    //I can't use on_byPlayer_applydebuff as it won't work in conjunction with on_byplayer_applydebuffstack
+    //so we do it this way, and just make a check if _currentStacks is 0, in which case we set it to 1 to indicate it already has been applied.
+    if (this._currentStacks === 0) {
+      this._currentStacks = 1;
+    }
+    this._currentStacks += 1;
+    if (this._currentStacks === 10) {
+      this.startOfMaxStacks = event.timestamp;
+    }
+  }
+  on_byPlayer_removedebuff(event) {
+    const debuffId = event.ability.guid;
+    if (debuffId !== SPELLS.TRUE_AIM_DEBUFF.id) {
+      return;
+    }
+    if (this._currentStacks === 10) {
+      this.timeAtMaxStacks += event.timestamp - this.startOfMaxStacks;
+    }
+    //ensures Trickshot cleaving doesn't count as multiple resets of stacks
+    if (this._currentStacks > 2) {
+      this.timesDropped += 1;
+    }
+    this._currentStacks = 0;
+  }
```

Comment:
Seems to me like it's missing adding to `timesDropped` if there were less than 3 stacks of the debuff (then it just resets the stacks).

Reply:
Take this example: 
![image](https://user-images.githubusercontent.com/29204244/31851796-762448ec-b66d-11e7-8789-8bc8f89b09a1.png)

There are only 4 meaningful drops (one on switching to maiden first time, one on back to avatar), then once gain once onto maiden and once onto Avatar. If I just let it accept all "removedebuff" events, it'll rocket up to 18 resets - this is due to trickshot cleaving applying a debuff on main target, then it'd cleave to target 2 and apply the debuff and thus remove it from the initial target. Sometimes you'll be casting 1 arcane shot, thus apply an extra true aim debuff, before it knocking it off when trick cleaving. 

I'll add a bit of information in the tooltip to clarify that:
 ``You reset your stacks (when they were at 3 stacks or higher) ${this.timesDropped} times over the course of the encounter. <br />Your total amount of resets (including with trickshot cleaving) was: ${this.totalTimesDropped}``
 - does this sound better?